packages/@react-spectrum/list/src/List.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {classNames, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DOMRef, Node} from '@react-types/shared';
+import {GridCollection, useGridState} from '@react-stately/grid';
+import {GridKeyboardDelegate, useGrid} from '@react-aria/grid';
+import {ListItem} from './ListItem';
+import {ListLayout} from '@react-stately/layout';
+import {ListState, useListState} from '@react-stately/list';
+import listStyles from './index.css';
+import React, {useMemo} from 'react';
+import {ReusableView} from '@react-stately/virtualizer';
+import {useCollator, useLocale} from '@react-aria/i18n';
+import {useProvider} from '@react-spectrum/provider';
+import {Virtualizer, VirtualizerItem} from '@react-aria/virtualizer';
+
+
+export const ListContext = React.createContext(null);
+
+export function useListLayout<T>(state: ListState<T>) {
+  let {scale} = useProvider();
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let layout = useMemo(() =>
+      new ListLayout<T>({
+        estimatedRowHeight: scale === 'large' ? 48 : 32,
+        estimatedHeadingHeight: scale === 'large' ? 33 : 26,
+        padding: 0, // TODO: get from DNA
+        collator
+      })
+    , [collator, scale]);
+
+  layout.collection = state.collection;
+  layout.disabledKeys = state.disabledKeys;
+  return layout;
+}
+
+function List<T extends object>(props, ref: DOMRef<HTMLDivElement>) {
+  let {
+    transitionDuration = 0
+  } = props;
+  let domRef = useDOMRef(ref);
+  let {collection} = useListState(props);
+
+  let gridCollection = new GridCollection({
+    columnCount: 1,
+    items: [...collection].map(item => ({
+      type: 'row',
+      childNodes: [{
+        ...item,
+        index: 0,
+        type: 'cell'
+      }]
+    }))
+  });
+
+
+  let state = useGridState({
+    ...props,
+    collection: gridCollection
+  });
+  let layout = useListLayout(state); // TODO should layout return nulls?
+
+  let collator = useCollator({usage: 'search', sensitivity: 'base'});
+  let {direction} = useLocale();
+  let keyboardDelegate = new GridKeyboardDelegate({
+    collection: state.collection,
+    disabledKeys: state.disabledKeys,
+    ref: domRef,
+    direction,
+    collator,
+    focusMode: 'cell'
+  });
+  let {gridProps} = useGrid({
+    keyboardDelegate,
+    ref: domRef
+  }, state);
```

Comment:
Noticed aria-label warnings when tabbing into the list:
![image](https://user-images.githubusercontent.com/9661127/114635935-b87bdb00-9c7a-11eb-9a27-ea4ef63076a4.png)

Probably need to pass `props` or at least the user provided `aria-label/labelledby` to `useGrid` here.